server/src/main/java/org/elasticsearch/action/ActionListener.java

```diff
@@ -191,7 +186,15 @@ public void onResponse(T t) {
 
         @Override
         public void onFailure(Exception e) {
-            bc.accept(delegate, e);
+            try {
+                bc.accept(delegate, e);
+            } catch (RuntimeException ex) {
+                if (ex != e) {
+                    ex.addSuppressed(e);
+                }
+                assert false : new AssertionError("map: listener.onResponse failed", ex);
```

Comment:
I think `map` is a leftover, since this is used from places not using ActionListener.map too, perhaps just:

```suggestion
                assert false : new AssertionError("listener.onResponse failed", ex);
```